### PR TITLE
New event: Fishing

### PR DESCRIPTION
**Pull request by NickTUD**
Event where you have to pretend you're fishing to complete it.

Because of the ScreenStore, it's **not possible** for me to test this class without refactoring the StrollEvent. Technically, this means this is not done yet. But I'd rather merge this into dev so that it includes this feature and test later. If anyone knows how to fix this, comment back.

Checkstyle, PMD, Findbugs seems fine in these classes, but you still need to check this.

### CODE REVIEW

File: core/src/cg/group4/game_logic/stroll/events/fishevent/CastForwardState.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package cg.group4.game_logic.stroll.events.fishevent;
+
+import com.badlogic.gdx.math.Vector3;
+
+/**
+ *
```
Comment (sirpolomotro) [inline core/src/cg/group4/game_logic/stroll/events/fishevent/CastForwardState.java]: Empty javadoc

File: core/src/cg/group4/game_logic/stroll/events/fishevent/FishingStrollEvent.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+	/**
+	 * The timer which keeps track for how long you hold still.
+	 */
+	protected Timer cFishTimer;
+
+	/**
+	 * Creates a new fishing event, with delay timer, text, screen and input.
+	 */
+	public FishingStrollEvent() {
+		super();
```
Comment (sirpolomotro) [inline core/src/cg/group4/game_logic/stroll/events/fishevent/FishingStrollEvent.java]: Volgens mij is dit overbodig in een constructor.

File: core/src/cg/group4/util/sensors/Accelerometer.java [hunk 3]
```diff
@@ -66,6 +66,7 @@ public final Vector3 update() {
         cBaseVector = readings;
         //System.out.println("Before filter: X: " + cBaseVector.x + " Y: " + cBaseVector.y + " Z: " + cBaseVector.z);
```
Comment (sirpolomotro) [inline core/src/cg/group4/util/sensors/Accelerometer.java]: weg gecommente code

File: core/test/cg/group4/game_logic/StandUpTest.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package cg.group4.game_logic;
+
+/**
+ * Created by Nick on 29-5-2015.
```
Comment (sirpolomotro) [inline core/test/cg/group4/game_logic/StandUpTest.java]: Empty test...

File: core/test/cg/group4/game_logic/stroll/StrollTest.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package cg.group4.game_logic.stroll;
+
+/**
```
Comment (sirpolomotro) [inline core/test/cg/group4/game_logic/stroll/StrollTest.java]: Empty test...

File: core/test/cg/group4/game_logic/stroll/events/TestStrollEventTest.java [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package cg.group4.game_logic.stroll.events;
+
+/**
+ * Created by Nick on 29-5-2015.
```
Comment (sirpolomotro) [inline core/test/cg/group4/game_logic/stroll/events/TestStrollEventTest.java]: Empty test...

File: core/test/cg/group4/game_logic/stroll/events/fishevent/FishingStrollEventTest.java [hunk 7]
```diff
@@ -0,0 +1,7 @@
+package cg.group4.game_logic.stroll.events.fishevent;
+
+/**
+ * Created by Nick on 29-5-2015.
```
Comment (sirpolomotro) [inline core/test/cg/group4/game_logic/stroll/events/fishevent/FishingStrollEventTest.java]: Empty test...

File: core/src/cg/group4/game_logic/stroll/events/fishevent/WaitState.java [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+    /**
+     * Method that processes the input to match the movement of waiting for the fish to bite, so holding still.
+     * @param input Vector containing the acceleration in the x,y,z directions respectively.
+     */
+    public final void processInput(final Vector3 input) {
+
+        float pythagorean = (float) Math.sqrt((Math.pow(input.x, 2) + Math.pow(input.y, 2) + Math.pow(input.z, 2)));
+        if (pythagorean > DELTA) {
+            cEvent.cFishTimer.reset();
```
Comment (sirpolomotro) [inline core/src/cg/group4/game_logic/stroll/events/fishevent/WaitState.java]: Als je deze timer vanuit de timerstore ophaalt dan kan cEvent gewoon een local variabele zijn

File: core/src/cg/group4/game_logic/stroll/Stroll.java [hunk 9]
```diff
@@ -113,7 +116,17 @@ protected final void generatePossibleEvent() {
         Random rnd = new Random();
         if (rnd.nextFloat() < cEventThreshold) {
             cEventGoing = true;
-            cEvent = new TestStrollEvent();
+            int chosenEvent = rnd.nextInt(2);
```
Comment (sirpolomotro) [inline core/src/cg/group4/game_logic/stroll/Stroll.java]: Couldn't you make this more dynamic by storing it in a list.. Maybe with the help of enums

**Comment by NickTUD** [comment]
Fixed some issues mentioned above

**Comment by sirpolomotro** [comment]
Not the nicest code but its okay for now..

**Comment by foresterre** [comment]
I've nothing to add to what Ben said.

**Pull request merged**